tests/components/ialarm/test_init.py

```diff
@@ -0,0 +1,65 @@
+"""Test the Antifurto365 iAlarm init."""
+from unittest.mock import Mock, patch
+from uuid import uuid4
+
+import pytest
+
+from homeassistant.components.ialarm import async_setup_entry, async_unload_entry
+from homeassistant.components.ialarm.const import DATA_COORDINATOR, DOMAIN
+from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from tests.common import MockConfigEntry
+
+TEST_DATA = {CONF_HOST: "192.168.10.20", CONF_PORT: 18034}
+
+
+async def init_integration(hass, entry_id):
+    """Do integration initialization."""
+    config_entry_mock = MockConfigEntry(
+        domain=DOMAIN,
+        data=TEST_DATA,
+        entry_id=entry_id,
+    )
+    return await async_setup_entry(hass, config_entry_mock)
+
+
+@pytest.fixture(name="ialarm_api")
+def ialarm_api_fixture():
+    """Set up IAlarm API fixture."""
+    with patch("homeassistant.components.ialarm.IAlarm") as mock_ialarm_api:
+        yield mock_ialarm_api
+
+
+async def test_setup_entry(hass, ialarm_api):
+    """Test setup entry."""
+    ialarm_api.return_value.get_mac = Mock(return_value="00:00:54:12:34:56")
+
+    entry_id = str(uuid4())
+    assert await init_integration(hass, entry_id)
+    ialarm_api.return_value.get_mac.assert_called_once()
+    assert hass.data[DOMAIN][entry_id][DATA_COORDINATOR] is not None
+
+
+async def test_setup_failed(hass, ialarm_api):
+    """Test setup failed due to an error."""
+    ialarm_api.return_value.get_mac = Mock(side_effect=ConnectionError)
+
+    entry_id = str(uuid4())
+    with pytest.raises(ConfigEntryNotReady):
+        await init_integration(hass, entry_id)
+
+
+async def test_unload_entry(hass):
+    """Test being able to unload an entry."""
+    entry_id = str(uuid4())
+    config_entry_mock = MockConfigEntry(
+        domain=DOMAIN,
+        data=TEST_DATA,
+        entry_id=entry_id,
+    )
+    hass.data.setdefault(DOMAIN, {})[entry_id] = {DATA_COORDINATOR: Mock}
+
+    assert entry_id in hass.data[DOMAIN]
+    await async_unload_entry(hass, config_entry_mock)
+    assert entry_id not in hass.data[DOMAIN]
```

Comment:
Don't call `async_unload_entry` or `async_setup_entry` use the standard interfaces in  https://github.com/home-assistant/core/blob/4cce724473233d4fb32c08bd251940b1ce2ba570/homeassistant/config_entries.py#L693

https://developers.home-assistant.io/docs/development_testing/#writing-tests-for-integrations

Reply:
I checked a couple of recently accepted PR and I changed this test accordingly. Now hopefully it should be okay.